cpp/src/tsne/cannylab/bh.cu

```diff
@@ -0,0 +1,1531 @@
+/*
```

Comment:
Just double-checking for the record - is this the only file from tsne-cuda that you used or modified?

Reply:
Yep that's the only file! Everything else (ie attractive forces, distance calculations, perplexity calculations are fully derived from scratch from Maaten's original paper + I did my own optimizations :) ). Only the repulsion kernels (ie those in bh.cu) were inspired from CannyLabs!